### PR TITLE
feat: add compact

### LINKED ISSUE #127: compact
https://lodash.com/docs/4.17.4#compact

### PR DESCRIPTION
**Pull request by char0n**
Closes #127

### CODE REVIEW

**Comment by char0n** [comment]
@guillaumearm do you have time to check it too ? Thanks

File: src/index.d.ts [hunk 1]
```diff
@@ -784,6 +784,12 @@ declare namespace RamdaAdjunct {
         */
         nonePass(predicates: Array<Function>): Function;
 
+        /**
+         * Creates an array with all falsy values removed.
+         * The values false, null, 0, "", undefined, and NaN are falsy.
+         */
+        compact<T>(list: Array<T>): Array<T>;
```
Comment (BjornMelgaard) [inline src/index.d.ts]: no, this is one of common mistakes (https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes, getMeAT section), this really should be `compact(list: Array<any>): Array<any>;`

**Comment by char0n** [comment]
Typings fixed. Thanks

**Comment by Undistraction** [comment]
Looks good.

**Pull request merged**